### PR TITLE
EIP: Move relevant metrics to CM

### PR DESCRIPTION
**Pull request by tssurya**
This is the metrics refactor for interconnect for the egressIP feature
following the work done in https://github.com/ovn-org/ovn-kubernetes/pull/3386.

**- What this PR does and why is it needed**

**- Special notes for reviewers**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: go-controller/pkg/metrics/cluster_manager.go [hunk 1]
```diff
@@ -54,6 +54,30 @@ var metricV6AllocatedHostSubnetCount = prometheus.NewGauge(prometheus.GaugeOpts{
 	Help:      "The total number of v6 host subnets currently allocated",
 })
 
+/** EgressIP metrics recorded from cluster-manager begins**/
+var metricEgressIPCount = prometheus.NewGauge(prometheus.GaugeOpts{
+	Namespace: MetricOvnkubeNamespace,
+	Subsystem: MetricOvnkubeSubsystemMaster,
```
Comment (martinkennelly) [inline go-controller/pkg/metrics/cluster_manager.go:60]: sub system needs to be changed to the new component.

File: go-controller/pkg/metrics/cluster_manager.go [hunk 2]
```diff
@@ -54,6 +54,30 @@ var metricV6AllocatedHostSubnetCount = prometheus.NewGauge(prometheus.GaugeOpts{
 	Help:      "The total number of v6 host subnets currently allocated",
 })
 
+/** EgressIP metrics recorded from cluster-manager begins**/
```
Comment (martinkennelly) [inline go-controller/pkg/metrics/cluster_manager.go:57]: Add the updates your doing to the metrics doc in /docs/. Downstream folks rely on that doc to adjust their testing  after metrics change.

File: go-controller/pkg/metrics/cluster_manager.go (hunk 1 above)
Comment (tssurya) [reply go-controller/pkg/metrics/cluster_manager.go:60]: ah good catch! done.

File: go-controller/pkg/metrics/cluster_manager.go (hunk 2 above)
Comment (tssurya) [reply go-controller/pkg/metrics/cluster_manager.go:57]: ack, let me do that ...

**Review by martinkennelly — APPROVED** [review]

**Pull request merged**